src/main/java/org/folio/innreach/controller/ContributionCriteriaController.java

```diff
@@ -0,0 +1,133 @@
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import java.util.HashSet;
+import java.util.UUID;
+
+@Log4j2
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/inn-reach/central-servers/")
+public class ContributionCriteriaController implements ContributionCriteriaApi {
+
+  private final ContributionCriteriaConfigurationService criteriaConfigurationService;
+  private final DateMapper dateMapper;
+
+  @Override
+  @PostMapping("/contribution-criteria")
+  public ResponseEntity<ContributionCriteriaDTO> postContributionCriteria(@Valid @RequestBody ContributionCriteriaDTO contributionCriteriaDTO) {
+    return ResponseEntity.status(HttpStatus.CREATED).body(toContributionCriteriaDTO(criteriaConfigurationService.create(toContributionCriteriaConfigurationDTO(contributionCriteriaDTO))));
+  }
+
+
+  @Override
+  @DeleteMapping("/{centralServerId}/contribution-criteria")
+  public ResponseEntity<Void> deleteCriteria(@PathVariable UUID centralServerId) {
+    criteriaConfigurationService.delete(centralServerId);
+    return ResponseEntity.noContent().build();
+  }
+
+  @Override
+  @GetMapping("/{centralServerId}/contribution-criteria")
+  public ResponseEntity<ContributionCriteriaDTO> getCriteriaById(@PathVariable UUID centralServerId) {
+    return ResponseEntity.ok(
+      toContributionCriteriaDTO(
+        criteriaConfigurationService.get(centralServerId)
+      )
+    );
+  }
+
+  @Override
+  @PutMapping("/{centralServerId}/contribution-criteria")
+  public ResponseEntity<ContributionCriteriaDTO> updateCriteria(@PathVariable UUID centralServerId,
+                                                            @Valid ContributionCriteriaDTO contributionCriteriaDTO) {
+    return ResponseEntity.ok(toContributionCriteriaDTO(
+      criteriaConfigurationService.update(toContributionCriteriaConfigurationDTO(contributionCriteriaDTO))
+    ));
+  }
+
+  public ContributionCriteriaDTO toContributionCriteriaDTO (ContributionCriteriaConfigurationDTO contributionCriteriaConfigurationDTO) {
+    var result = new ContributionCriteriaDTO();
+    result.setCentralServerId(contributionCriteriaConfigurationDTO.getCentralServerId());
+    var metadata = new Metadata();
+    metadata.setCreatedByUserId(contributionCriteriaConfigurationDTO.getCreatedBy());
+    metadata.setUpdatedByUserId(contributionCriteriaConfigurationDTO.getLastModifiedBy());
+    metadata.setCreatedDate(dateMapper.offsetDateTimeAsDate(contributionCriteriaConfigurationDTO.getCreatedDate()));
+    metadata.setUpdatedDate(dateMapper.offsetDateTimeAsDate(contributionCriteriaConfigurationDTO.getLastModifiedDate()));
+    result.setMetadata(metadata);
+    contributionCriteriaConfigurationDTO.getExcludedLocations().stream().forEach(excludedLocationDTO -> result.addLocationIdsItem(excludedLocationDTO.getExcludedLocationId()));
```

Comment:
Long methods like this are bad practice, I would break them down into smaller separate methods, also consider a different approach instead of switch-case. All this leads to "spaghetti code".